src/unitxt/operators.py

```diff
@@ -213,6 +213,103 @@ def process(
         return flatten_dict(instance, parent_key=self.parent_key, sep=self.sep)
 
 
+class ModelInputFormatter(StreamInstanceOperator):
+    r"""Generates the whole input to the model, from constant strings that are given as args, and from values found in specified fields of the instance.
+
+    WholeInputFormatter expects its input instance to have been processed by a Template StreamInstanceOperator
+    (from templates.py), or any extension thereof, and hence to contain a field named "source", and to have
+    been processed by RenderInstruction StreamInstanceOperator (from renderer.py) and hence to contain a non None
+    field "instruction".
+    Demos are assumed to be included as well, in a field whose name is specified by argument 'demos_field'. That
+    field is assumed to contain a list of dicts, each containing keys: "source" and "target" of a single demo.
+    And finally, system_prompt is also assumed to be included
+    in the instance, in a field whose name is specified by argument 'system_prompt'.
+    The value in field "source" of the input instance describes (verbalizes) original values in the instance (as read
+    from the source dataset), in the context of the underlying task.
+    WholeInputFormatter makes this (verbalization of the) input into a whole-input-to-the-model, by combining it
+    with instruction, demos, and system_prompt, read from the input instance, through both formatting args
+    formatting arguments.
+    Following the Renderers action, the formatted model input is fed into field "source" (overwriting the field)
+    Also following renderers, this operator pops fields "instruction" and "demos" from the input instance.
+
+    Args:
+        demos_field (str): the name of the field that contains the demos, being dicts with "source" and "target" keys
+        system_prompt (str): the name of the field containing an introductory text and or xml tags
+        demo_format (str): formatting string for a single demo, combining fields source and target
+        model_input_format (str) overall output format, combining system_prompt, instruction, demos, and source
+    """
+
+    demos_field: str = "demos"
+    system_prompt: str = None
```

Comment:
```suggestion
    system_prompt_field: str = None
```

Reply:
since its a field name, AFAIU

Reply:
Checking and testing now, `system_prompt` seems an orphan field. No one uses it, and if @yoavkatz agrees that indeed orphan, I will remove that arg altogether.

Reply:
There is some terminology mismatch now.   

System prompt - is what us today called instruction.

The "Instruction" which should be coming from the template (like source and target) , is not yet implemented.

Reply:
I see. Removing `system_prompt`    והיה כלא היה